### PR TITLE
update code.org/translate to use list of languages from cdo-languages

### PR DESCRIPTION
**Pull request by tanyaparker**

### CODE REVIEW

**Comment by tanyaparker** [comment]
@wjordan suggested using lib/cdo instead of chef since I'm looking to use this list for views and not for server side stuff. I mimicked the behavior of properties.rb like how we use fetch_hoc_metrics throughout the site. With this method I can just update the cdo-languages gsheet, and it should be able to be used for everything pegasus and my i18n scripts.

I'm not sure if this still adds stress to the DB? If it's still a concern, I could just have languages be a hash/json, update anywhere else using the DB list to use this, and remove cdo-languages as a DB table.

File: pegasus/sites.v3/code.org/views/target_languages.haml [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-
 #language{:style=>'width: 780px; padding: 0; margin: 0;'}
-  - languages.each_slice(6) do |group|
+  - fetch_crowdin_languages.each_slice(6) do |group|
     .language_row{:style=>'width: 780px; padding: 0; margin: 0;'}
-      - group.each do |name, code|
+      - group.each do |properties|
```
Comment (mehalshah) [inline pegasus/sites.v3/code.org/views/target_languages.haml:4]: If you are only interacting with one element of properties, then might as well enumerate through that list

Something like

group.collect {|g| g[:crowdin_code_s]}.each

Though I think there's a better way to write that

File: lib/cdo/languages.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  @@table = DB[:cdo_languages]
+
+  def self.get_crowdin_languages()
+    @@table.select(:crowdin_code_s, :crowdin_name_s).where("crowdin_code_s != 'en'")
+  end
+
+end
+
+def fetch_crowdin_languages()
```
Comment (mehalshah) [inline lib/cdo/languages.rb]: Not sure why this is needed - you can just call Languages.get_crowdin_languages in target_languages.haml, right?

**Comment by mehalshah** [comment]
I think this is fine - it will load the languages on startup and not hit it again.

One comment, one question, the rest LGTM

File: pegasus/sites.v3/code.org/views/target_languages.haml (hunk 1 above)
Comment (tanyaparker) [inline pegasus/sites.v3/code.org/views/target_languages.haml:4]: I have both crowdin_code_s and crowdin_name_s

Comment (mehalshah) [inline pegasus/sites.v3/code.org/views/target_languages.haml:4]: Oh then ignore this

On Mon, Mar 7, 2016 at 5:49 PM, Tanya Parker notifications@github.com
wrote:

> In pegasus/sites.v3/code.org/views/target_languages.haml
> https://github.com/code-dot-org/code-dot-org/pull/7182#discussion_r55305373
> :
> 
> > ```
> >  .language_row{:style=>'width: 780px; padding: 0; margin: 0;'}
> > ```
> > -      - group.each do |name, code|
> > -      - group.each do |properties|
> 
> I have both crowdin_code_s and crowdin_name_s
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/code-dot-org/code-dot-org/pull/7182/files#r55305373.

## 

Mehal Shah

**Comment by joshlory** [comment]
@tanyaparker I think this still queries the DB each page load (since the `@@table` ref is cached but not the result of the query).

**Comment by tanyaparker** [comment]
@joshlory I thought it would only call the DB once? And then all the other calls would refer to `@@table`? What would I have to do differently so it doesn't call DB each time?

**Pull request merged**